plugins/kubernetes/src/utils/clusterLinks/formatClusterLink.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { defaultFormatterName, clusterLinksFormatters } from './formatters';
+
+export function formatClusterLink(options: {
+  dashboardUrl?: string;
+  dashboardApp?: string;
+  object: any;
+  kind: string;
+}) {
+  if (!options.dashboardUrl) {
+    return undefined;
+  }
+  if (!options.object) {
+    return options.dashboardUrl;
+  }
+  const app = options.dashboardApp || defaultFormatterName;
+  const formatter = clusterLinksFormatters[app];
+  if (!formatter) {
+    throw new Error(`Could not find Kubernetes dashboard app named '${app}'`);
```

Comment:
if anything fails in here (formatter missing, formatter throws etc), should that error really be rethrown? should we just return undefined or the unmodified dashboardUrl or something instead? not sure it's a good thing to explode the app if just some entity misconfig has happened

Reply:
Added a Warning panel when there is an error, instead of crashing the whole tab. This is definitively a better behavior!